scipy/integrate/tests/test__quad_vec.py

```diff
@@ -123,3 +144,35 @@ def f_inf(x):
 
     res, err, info = quad_vec(f_inf, 0, 1, full_output=True)
     assert info.status == 3
+
+
+@pytest.mark.parametrize('a,b', [(0, 1), (0, np.inf), (np.inf, 0),
+                                 (-np.inf, np.inf), (np.inf, -np.inf)])
+def test_points(a, b):
+    # Check that initial interval splitting is done according to
+    # `points`, by checking that consecutive sets of 15 point (for
+    # gk15) function evaluations lie between `points`
+
+    points = (0, 0.25, 0.5, 0.75, 1.0)
+    points += tuple(-x for x in points)
+
+    quadrature_points = 15
+    interval_sets = []
+    count = 0
+
+    def f(x):
+        nonlocal count
```

Comment:
(Minor, only worth addressing together with something else, if any) can this be attached to the function as an attribute instead? `f.count` seems less magical then a `nonlocal` statement.